### PR TITLE
Add in the NanoLyse process

### PR DESCRIPTION
**Pull request by yuukiiwa**
Hi @drpatelh, I have added in the `NANOLYSE` process and the `--skip_control_cleaning` flag. Thanks!

### CODE REVIEW

File: main.nf [hunk 1]
```diff
@@ -518,6 +523,42 @@ if (!params.skip_basecalling) {
...
+
+        input:
+        tuple val(sample), path(fastq) from ch_fastq_nanolyse.map{ ch -> [ ch[0], ch[1] ] }
+
+        output:
+        tuple val(sample), path("*.fastq.gz") into ch_nanolyse_fastq
+
+        script:
+        sample_fastq_gz = "$sample"+".fastq.gz"
+        lambda_fasta_gz = "lambda.fasta.gz"
```
Comment (drpatelh) [inline main.nf:543]:
```suggestion
```

File: main.nf [hunk 2]
```diff
@@ -518,6 +523,42 @@ if (!params.skip_basecalling) {
...
+     * Control cleaning using nanolyse
+     */
+    process NANOLYSE {
+        echo true
+        label 'process_medium'
+        publishDir "${params.outdir}/nanolyse", mode: params.publish_dir_mode
+
+        input:
+        tuple val(sample), path(fastq) from ch_fastq_nanolyse.map{ ch -> [ ch[0], ch[1] ] }
+
```
Comment (drpatelh) [inline main.nf:537]:
```suggestion
        path fasta from ch_nanolyse_fasta
```

File: nextflow.config [hunk 3]
```diff
@@ -29,6 +29,7 @@ params {
   qcat_detect_middle         = false
   skip_basecalling           = false
   skip_demultiplexing        = false
+  skip_control_cleaning      = false
```
Comment (drpatelh) [inline nextflow.config:32]:
```suggestion
  run_nanolyse               = false
```

File: nextflow_schema.json [hunk 4]
```diff
@@ -135,6 +135,11 @@
                     "type": "boolean",
                     "description": "Skip demultiplexing with Guppy/qcat.",
                     "fa_icon": "fas fa-fast-forward"
+                },
+                "skip_control_cleaning": {
```
Comment (drpatelh) [inline nextflow_schema.json:139]:
```suggestion
                "run_nanolyse": {
```

File: nextflow_schema.json [hunk 5]
```diff
@@ -135,6 +135,11 @@
...
+                    "type": "boolean",
+                    "description": "Skip control cleaning with NanoLyse.",
```
Comment (drpatelh) [inline nextflow_schema.json:141]:
```suggestion
                    "description": "Filter reads from FastQ files mapping to the lambda phage genome using NanoLyse",
```

File: main.nf [hunk 6]
```diff
@@ -1241,6 +1282,7 @@ process GET_SOFTWARE_VERSIONS {
     echo $workflow.manifest.version > v_pipeline.txt
     echo $workflow.nextflow.version > v_nextflow.txt
     qcat --version &> v_qcat.txt
+    NanoLyse --version &> v_nanolyse.txt
```
Comment (drpatelh) [inline main.nf:1285]: Need to add this here too https://github.com/nf-core/nanoseq/blob/ad5b2bb7a37a6edb9f61603636bd26e43259f58b/bin/scrape_software_versions.py#L11

**Review by drpatelh — CHANGES_REQUESTED** [review]

**Comment by drpatelh** [comment]
Looks great overall @yuukiiwa ! Be great if we can rename `--skip_control_cleaning` to `--run_nanolyse`. This means it isn't run by default but can if required. Also, I think we should add another option to provide the fasta file passed to Nanolyse as a parameter i.e. `--nanolyse_fasta`. Will need to check if this exists if `--run_nanolyse` is specified and stage it properly to pass it to the process. Sorry, it's taken this long to get to it. 

PS: You will need to pull in the latest changes from `dev` because I updated the pipeline with the nf-core pipeline template updates.

File: main.nf [hunk 7]
```diff
@@ -518,6 +523,42 @@ if (!params.skip_basecalling) {
...
+    /*
+     * Control cleaning using nanolyse
+     */
+    process NANOLYSE {
+        echo true
+        label 'process_medium'
+        publishDir "${params.outdir}/nanolyse", mode: params.publish_dir_mode
+
+        input:
+        tuple val(sample), path(fastq) from ch_fastq_nanolyse.map{ ch -> [ ch[0], ch[1] ] }
```
[the remark this answers is not in the document]
Comment (yuukiiwa) [reply main.nf:536]: It uses only the sample and fastq from the ch_fastq_nanolyse channel.

File: main.nf (hunk 6 above)
Comment (yuukiiwa) [reply main.nf:1325]: I have added a similar line for the NanoLyse process.

**Comment by drpatelh** [comment]
Great job @yuukiiwa !

**Review by drpatelh — APPROVED** [review]

**Pull request merged**